internal/pkg/server/server.go

```diff
@@ -82,6 +120,201 @@ func (s *Server) postTar(c echo.Context) error {
 	return s.streamImages(c, authn, normalizeImages(images))
 }
 
+func (s *Server) postHelmRepo(c echo.Context) error {
+	name := c.FormValue("name")
+	url := c.FormValue("url")
+	if name == "" {
+		return fmt.Errorf("repository name not set")
+	}
+	if url == "" {
+		return fmt.Errorf("repository URL not set")
+	}
+
+	// The following code is adapted from https://github.com/helm/helm/blob/master/cmd/helm/repo_add.go
+	fileLock := flock.New(strings.Replace(repoFile, filepath.Ext(repoFile), ".lock", 1))
+	lockCtx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+	locked, err := fileLock.TryLockContext(lockCtx, time.Second)
+	if err == nil && locked {
+		defer fileLock.Unlock()
+	}
+	if err != nil {
+		return err
+	}
+
+	b, err := ioutil.ReadFile(repoFile)
+	if err != nil && !os.IsNotExist(err) {
+		return err
+	}
+
+	var f repo.File
+	if err := yaml.Unmarshal(b, &f); err != nil {
+		return err
+	}
+
+	cfg := repo.Entry{
+		Name:                  name,
+		URL:                   url,
+		Username:              c.FormValue("username"),
+		Password:              c.FormValue("password"),
+		InsecureSkipTLSverify: false,
+	}
+	repo, err := repo.NewChartRepository(&cfg, getters)
+	if err != nil {
+		return err
+	}
+	if repoCache != "" {
+		repo.CachePath = repoCache
+	}
+
+	if _, err := repo.DownloadIndexFile(); err != nil {
+		return errors.Wrapf(err, "looks like %q is not a valid chart repository or cannot be reached", url)
```

Comment:
```suggestion
		return fmt.Errorf("looks like %q is not a valid chart repository or cannot be reached: %w", url, err)
```
Go has now a native way to wrap errors which I prefer. See [fmt.Errorf](https://golang.org/pkg/fmt/#Errorf) and [errors](https://golang.org/pkg/errors/).